pynumdiff/linear_model/_linear_model.py

```diff
@@ -1,115 +1,25 @@
-import copy, math, logging, scipy
+import math, scipy
 import numpy as np
 from warnings import warn
 
-from pynumdiff.finite_difference import first_order as finite_difference
+from pynumdiff.finite_difference import second_order as finite_difference
+from pynumdiff.polynomial_fit import savgoldiff as _savgoldiff # patch through
+from pynumdiff.polynomial_fit import polydiff as _polydiff # patch through
 from pynumdiff.utils import utility
 
 try: import cvxpy
 except ImportError: pass
 
 
-#########################
-# Savitzky-Golay filter #
-#########################
-def savgoldiff(x, dt, params=None, options=None, poly_order=None, window_size=None, smoothing_win=None):
-    """Use the Savitzky-Golay to smooth the data and calculate the first derivative. It uses
-    scipy.signal.savgol_filter. The Savitzky-Golay is very similar to the sliding polynomial fit,
-    but slightly noisier, and much faster.
+def savgoldiff(*args, **kwargs):
+    warn("`savgoldiff` has moved to `polynomial_fit.savgoldiff` and will be removed from "
```

Comment:
This is just a common pattern to make the functions still importable from the old place, for backwards compatibility.